### ISSUE TITLE
Allow pane_grid to be empty

### ISSUE THREAD

**Issue by wiiznokes** [iced-rs/iced#2038]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

in pane_grid, there is this method:
```rust
/// Returns `true` if the amount of panes in the [`State`] is 0.
pub fn is_empty(&self) -> bool {
    self.len() == 0
}
```
but rn, it will never return true, because the `close` method doesn't remove the pane if there is only one.

### What is the expected behavior?

`close` method should remove the pane, even if there is only one.
For that, there should have some more methods:
- `new_empty` to allow the creation of an empty pane_grid
- `add_pane` because `split` method will not work on an empty pane.

### Version

crates.io release

### Operating System

Windows

**Comment by Redhawk18** [comment]
I agree, I just added this widget to my application. How my widget works is if my data struct is empty the widget doesn't show. This all works correctly expect pane grid shows a empty pane, and the only way to get around this is to just add a conditional in my `view`.
